src/palace.pyx

```diff
@@ -486,11 +509,53 @@ cdef class Context:
         return self.impl.is_supported(to_channel_config(channel_config),
                                       to_sample_type(sample_type))
 
-    # TODO: available resamplers
-    # TODO: default resampler index
+    @property
+    def available_resamplers(self) -> List[str]:
+        """Query the list of resamplers supported by the context.
+
+        If the `AL_SOFT_source_resampler` extension is unsupported
+        this will be an empty list, otherwise there would be
+        at leaste one entry.
+        """
+        cdef alure.ArrayView[string] resamplers
+        resamplers = self.impl.get_available_resamplers()
+        return [resampler for resampler in resamplers]
+
+    @property
+    def default_resampler_index(self) -> int:
+        """The context's default resampler index.
+
+        If the `AL_SOFT_source_resampler` extension is unsupported
+        the resampler list will be empty and this will return 0.
+
+        If you try to access the resampler list with this index
+        without extension, undefined behavior will occur
+        (accessing an out of bounds array index).
+        """
+        return self.impl.get_default_resampler_index()
+
     # TODO: async buffers
-    # TODO: doppler factor
-    # TODO: speed of sound
+
+    @setter
+    def doppler_factor(self, factor: float) -> None:
+        """The doppler factor to apply to all source
+        doppler calculations.
```

Comment:
Use `value` here.  Also by PEP 257, it's preferable to fit the first paragraph of docs in a single line, e.g. *factor to apply to all source's doppler calculations*.